Include/ceval.h

```diff
@@ -86,48 +86,39 @@ PyAPI_FUNC(int) Py_MakePendingCalls(void);
 PyAPI_FUNC(void) Py_SetRecursionLimit(int);
 PyAPI_FUNC(int) Py_GetRecursionLimit(void);
 
-#if !PY_DEBUGGING_CHECKS
-#define Py_EnterRecursiveCall(where) (0)
-#define Py_LeaveRecursiveCall(where) ((void)0)
-#else
 #define Py_EnterRecursiveCall(where)  \
-            (_Py_MakeRecCheck(PyThreadState_GET()->recursion_depth) &&  \
+            (_Py_MakeRecCheck(PyThreadState_GET()->stack_limit) &&  \
              _Py_CheckRecursiveCall(where))
-#define Py_LeaveRecursiveCall()                         \
-    do{ if(_Py_MakeEndRecCheck(PyThreadState_GET()->recursion_depth))  \
-      PyThreadState_GET()->overflowed = 0;  \
-    } while(0)
-#endif
+#define Py_LeaveRecursiveCall() ((void)0)
 PyAPI_FUNC(int) _Py_CheckRecursiveCall(const char *where);
 
 /* Due to the macros in which it's used, _Py_CheckRecursionLimit is in
    the stable ABI.  It should be removed therefrom when possible.
 */
 PyAPI_DATA(int) _Py_CheckRecursionLimit;
-PyAPI_DATA(int) _Py_RecursionLimitLowerWaterMark_Precomputed;
+
+static inline void* _stack_pointer() {
+    void* sp;
+    __asm( "mov %%rsp, %0" : "=rm" ( sp ));
+    return sp;
+}
+// This could also be defined as __builtin_frame_address(0) for a
+// bit better portability. But that uses rbp, which means that
+// any code calling EnterRecursiveCall will end up saving rbp,
+// and so using rsp frees up the rbp register
+#define frame_address() _stack_pointer()
 
 #ifdef USE_STACKCHECK
 /* With USE_STACKCHECK, trigger stack checks in _Py_CheckRecursiveCall()
    on every 64th call to Py_EnterRecursiveCall.
 */
-#  define _Py_MakeRecCheck(x)  \
-    (++(x) > _Py_CheckRecursionLimit || \
+#  define _Py_MakeRecCheck()  \
+    (frame_address() < (x) || \
```

Comment:
I think this code will not work because of the removal of `x` argument.
I'm also fine with just raising a explicit compile error if USE_STACKCHECK is defined but we have not tested it.